examples/transformer_base.py

```diff
@@ -104,7 +104,8 @@ def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx, second_orde
         self.lr_scheduler.step()
 
     def get_tqdm_dict(self):
-        tqdm_dict = {"loss": "{:.3f}".format(self.trainer.avg_loss), "lr": self.lr_scheduler.get_last_lr()[-1]}
+        avg_loss = getattr(self.trainer, "avg_loss", 0.0)
```

Comment:
@williamFalcon 
Is there a cleaner way to fix this?

Reply:
i think this is outdated. We should deprecate  this function.

Instead use:

```python

bar = {...}
return {"loss": loss, "progress_bar": bar}
```